### PR TITLE
Devcontainer for Contributors

### PR DESCRIPTION
**Pull request by stephenandary**
Looking to add a devcontainer to make contributing to dioxus easier. With a devcontainer a contributor doesn't need to worry about dependencies as long as code is built and run within the container which can be done locally through a vscode extension or in codespaces which I believe is free for OSS. 

Issues may arise if using gpg signing, but vscode tries to do a good job about syncing keys through vscode. If it is an issue user can just reopen the directory locally with a few clicks to commit. 

Some links: 
https://code.visualstudio.com/docs/devcontainers/containers
https://containers.dev/
https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers

### CODE REVIEW

File: .devcontainer/local-features/linux/install.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+echo "Update Apt Database"
+sudo apt-get update
+
+echo "Install Webkit Dependencies"
+sudo apt-get -qq install build-essential libwebkit2gtk-4.0-dev libgtk-3-dev libayatana-appindicator3-dev
```
Comment (stephenandary) [inline .devcontainer/local-features/linux/install.sh:6]: libwebkit2gtk-4.1-dev doesn't seem to be found when trying to install

when running cargo build in devcontainer or locally, get many warnings for name conflicts and some errors such as glib being less than version 2.70. 

will need to touch base with someone about the dependency needs for this. Currently getting info from workflow here: 
https://github.com/DioxusLabs/dioxus/blob/master/.github/workflows/main.yml#L41

**Comment by stephenandary** [comment]
Getting warnings with rustlang/rust image that I didn't get with rust base image: 
```
warning: output filename collision.
The example target `simple` in package `dioxus-router v0.3.0 (/workspaces/dioxus/packages/router)` has the same output filename as the example target `simple` in package `dioxus-native-core v0.2.0 (/workspaces/dioxus/packages/native-core)`.
Colliding filename is: /workspaces/dioxus/target/debug/examples/simple
The targets should have unique names.
Consider changing their names to be unique or compiling them separately.
This may become a hard error in the future; see <https://github.com/rust-lang/cargo/issues/6313>.
warning: output filename collision.
The example target `simple` in package `dioxus-router v0.3.0 (/workspaces/dioxus/packages/router)` has the same output filename as the example target `simple` in package `dioxus-native-core v0.2.0 (/workspaces/dioxus/packages/native-core)`.
Colliding filename is: /workspaces/dioxus/target/debug/examples/simple.dwp
The targets should have unique names.
Consider changing their names to be unique or compiling them separately.
This may become a hard error in the future; see <https://github.com/rust-lang/cargo/issues/6313>.
warning: output filename collision.
The example target `all_events` in package `dioxus-tui v0.2.2 (/workspaces/dioxus/packages/dioxus-tui)` has the same output filename as the example target `all_events` in package `dioxus-examples v0.0.0 (/workspaces/dioxus)`.
Colliding filename is: /workspaces/dioxus/target/debug/examples/all_events
The targets should have unique names.
Consider changing their names to be unique or compiling them separately.
This may become a hard error in the future; see <https://github.com/rust-lang/cargo/issues/6313>.
warning: output filename collision.
The example target `all_events` in package `dioxus-tui v0.2.2 (/workspaces/dioxus/packages/dioxus-tui)` has the same output filename as the example target `all_events` in package `dioxus-examples v0.0.0 (/workspaces/dioxus)`.
Colliding filename is: /workspaces/dioxus/target/debug/examples/all_events.dwp
The targets should have unique names.
Consider changing their names to be unique or compiling them separately.
This may become a hard error in the future; see <https://github.com/rust-lang/cargo/issues/6313>.
warning: output filename collision.
The example target `components` in package `dioxus-tui v0.2.2 (/workspaces/dioxus/packages/dioxus-tui)` has the same output filename as the example target `components` in package `dioxus-guide v0.0.1 (/workspaces/dioxus/docs/guide)`.
Colliding filename is: /workspaces/dioxus/target/debug/examples/components
The targets should have unique names.
Consider changing their names to be unique or compiling them separately.
This may become a hard error in the future; see <https://github.com/rust-lang/cargo/issues/6313>.
warning: output filename collision.
The example target `components` in package `dioxus-tui v0.2.2 (/workspaces/dioxus/packages/dioxus-tui)` has the same output filename as the example target `components` in package `dioxus-guide v0.0.1 (/workspaces/dioxus/docs/guide)`.
Colliding filename is: /workspaces/dioxus/target/debug/examples/components.dwp
The targets should have unique names.
Consider changing their names to be unique or compiling them separately.
This may become a hard error in the future; see <https://github.com/rust-lang/cargo/issues/6313>.
warning: output filename collision.
The example target `readme` in package `dioxus-tui v0.2.2 (/workspaces/dioxus/packages/dioxus-tui)` has the same output filename as the example target `readme` in package `dioxus-examples v0.0.0 (/workspaces/dioxus)`.
Colliding filename is: /workspaces/dioxus/target/debug/examples/readme
The targets should have unique names.
Consider changing their names to be unique or compiling them separately.
This may become a hard error in the future; see <https://github.com/rust-lang/cargo/issues/6313>.
warning: output filename collision.
The example target `readme` in package `dioxus-tui v0.2.2 (/workspaces/dioxus/packages/dioxus-tui)` has the same output filename as the example target `readme` in package `dioxus-examples v0.0.0 (/workspaces/dioxus)`.
Colliding filename is: /workspaces/dioxus/target/debug/examples/readme.dwp
The targets should have unique names.
Consider changing their names to be unique or compiling them separately.
This may become a hard error in the future; see <https://github.com/rust-lang/cargo/issues/6313>.
warning: output filename collision.
The example target `stress` in package `dioxus-tui v0.2.2 (/workspaces/dioxus/packages/dioxus-tui)` has the same output filename as the example target `stress` in package `dioxus v0.3.2 (/workspaces/dioxus/packages/dioxus)`.
Colliding filename is: /workspaces/dioxus/target/debug/examples/stress
The targets should have unique names.
Consider changing their names to be unique or compiling them separately.
This may become a hard error in the future; see <https://github.com/rust-lang/cargo/issues/6313>.
warning: output filename collision.
The example target `stress` in package `dioxus-tui v0.2.2 (/workspaces/dioxus/packages/dioxus-tui)` has the same output filename as the example target `stress` in package `dioxus v0.3.2 (/workspaces/dioxus/packages/dioxus)`.
Colliding filename is: /workspaces/dioxus/target/debug/examples/stress.dwp
The targets should have unique names.
Consider changing their names to be unique or compiling them separately.
This may become a hard error in the future; see <https://github.com/rust-lang/cargo/issues/6313>.
warning: output filename collision.
The example target `counter` in package `rink v0.2.2 (/workspaces/dioxus/packages/rink)` has the same output filename as the example target `counter` in package `dioxus-examples v0.0.0 (/workspaces/dioxus)`.
Colliding filename is: /workspaces/dioxus/target/debug/examples/counter
The targets should have unique names.
Consider changing their names to be unique or compiling them separately.
This may become a hard error in the future; see <https://github.com/rust-lang/cargo/issues/6313>.
warning: output filename collision.
The example target `counter` in package `rink v0.2.2 (/workspaces/dioxus/packages/rink)` has the same output filename as the example target `counter` in package `dioxus-examples v0.0.0 (/workspaces/dioxus)`.
Colliding filename is: /workspaces/dioxus/target/debug/examples/counter.dwp
The targets should have unique names.
Consider changing their names to be unique or compiling them separately.
This may become a hard error in the future; see <https://github.com/rust-lang/cargo/issues/6313>.
```

**Comment by Demonthos** [comment]
> Getting warnings with rustlang/rust image:

Those errors are just lints for the workspace. It shouldn't be an issue for this PR. Most likely one package has a newer set of lints. Feel free to open a new issue with the command that triggers this, and I can go through renaming the duplicate tests/examples.

**Comment by jkelleyrtp** [comment]
Awesome - thank you!

**Pull request merged**